wagon.py

```diff
@@ -0,0 +1,1015 @@
+    pip_command = _construct_pip_command(
+        package,
+        wheels_path,
+        virtualenv,
+        requirements_file,
+        upgrade,
+        install_args)
+
+    if IS_VIRTUALENV and not virtualenv:
+        logger.info('Installing within current virtualenv: {0}...'.format(
+            IS_VIRTUALENV))
+
+    result = run(pip_command)
+    if not result.returncode == 0:
+        raise WagonError('Could not install package: {0} ({1})'.format(
+            package, result.aggr_stderr))
+
+
+def _get_downloaded_wheels(path):
+    return \
+        [filename for filename in os.listdir(path) if filename.endswith('whl')]
+
+
+def _download_file(url, destination):
+    logger.info('Downloading {0} to {1}...'.format(url, destination))
+    final_url = urllib.urlopen(url).geturl()
+    if final_url != url:
+        logger.debug('Redirected to {0}'.format(final_url))
+    f = urllib.URLopener()
+    f.retrieve(final_url, destination)
+
+
+def _zip(source, destination):
+    logger.info('Creating zip archive: {0}...'.format(destination))
+    with closing(zipfile.ZipFile(destination, 'w')) as zip:
+        for root, _, files in os.walk(source):
+            for filename in files:
+                file_path = os.path.join(root, filename)
+                source_dir = os.path.dirname(source)
+                zip.write(file_path, os.path.relpath(file_path, source_dir))
+
+
+def _unzip(archive, destination):
+    logger.debug('Extracting zip {0} to {1}...'.format(archive, destination))
+    with closing(zipfile.ZipFile(archive, 'r')) as zip:
+        zip.extractall(destination)
+
+
+def _tar(source, destination):
+    logger.info('Creating tgz archive: {0}...'.format(destination))
+    with closing(tarfile.open(destination, 'w:gz')) as tar:
+        tar.add(source, arcname=os.path.basename(source))
+
+
+def _untar(archive, destination):
+    logger.debug('Extracting tgz {0} to {1}...'.format(archive, destination))
+    with closing(tarfile.open(name=archive)) as tar:
+        tar.extractall(path=destination, members=tar.getmembers())
+
+
+def _get_wheel_tags(wheel_name):
+    filename, _ = os.path.splitext(os.path.basename(wheel_name))
+    return filename.split('-')
+
+
+def _get_platform_from_wheel_name(wheel_name):
+    """Extracts the platform of a wheel from its file name.
+    """
+    logger.error('AAAAAAA:{0}'.format(wheel_name))
+    return _get_wheel_tags(wheel_name)[-1]
+
+
+def _get_platform_for_set_of_wheels(wheels_path):
+    """For any set of wheel files, extracts a single platform.
+
+    Since a set of wheels created or downloaded on one machine can only
+    be for a single platform, if any wheel in the set has a platform
+    which is not `any`, it will be used. If a platform other than
+    `any` was not found, `any` will be assumed.
+    """
+    for wheel in _get_downloaded_wheels(wheels_path):
+        platform = _get_platform_from_wheel_name(
+            os.path.join(wheels_path, wheel))
+        if platform != 'any':
```

Comment:
IMO, 'any' is a magic word. maybe it would be better to assign it to a constant value that would reveal its meaning in this context. something like:
ALL_PLATFORMS = 'any'
and then the code would read: if wheel_platform != ALL_PLATFORMS